frontend/src/ui/pages/RegisterPage/RegisterPage.js

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useState } from 'react'
+import './RegisterPage.scss'
+import { Formik, Form, Field, ErrorMessage } from 'formik'
+import * as Yup from 'yup'
+import { Button, Text } from '../../components'
+import { register } from '../../../ducks/UserApi'
+import { useNavigate } from 'react-router'
+import RegisterPopup from './RegisterPopup'
+import { Link } from 'react-router-dom'
+
+function RegisterPage() {
+
+	const navigate = useNavigate()
+
+	const [currentBody, setCurrentBody] = useState(0)
+	const [currentHair, setCurrentHair] = useState(0)
+	const [currentFit, setCurrentFit] = useState(0)
+	const [toggle, setToggle] = useState(false)
+
+	const [errors, setErrors] = useState([])
+    
+	const registerSchema = Yup.object().shape({
+		password: Yup.string().required('Wait! What\'s your password?'),
+		passwordConfirmation: Yup.string().required('Confirm your passowrd.').oneOf([Yup.ref('password'), null], 'Passwords don\'t match.'),
+		username: Yup.string().required('Wait! What\'s your username?')
+	})
+
+	const handleSubmit = async (values) => {
+		const avatar = {
+			'body': /^([^.]+)/.exec(Object.keys(body)[currentBody])[1],
+			'hair': /^([^.]+)/.exec(Object.keys(hair)[currentHair])[1],
+			'fit': /^([^.]+)/.exec(Object.keys(fit)[currentFit])[1]
+		}
+
+		const user = {...values, hairID: parseInt(avatar.hair),  outfitTopID: parseInt(avatar.fit), outfitBottomID: parseInt(avatar.body)}
+
+		const res = await register(user)
+		if (!res) {
+			setErrors(['Server not available'])
+		} else {
+			if (res.status === 201) {
+				setToggle(true)
+				setTimeout(() => {
+					navigate('/login')
+				}, 3000)
+			} else if (res.status === 400) {
+				setTimeout(() => {
+					setErrors([])
+				}, 4000)
+				setErrors([res.data.message, ...errors])
+			}
+		}
+	}
+
+	function importAll(r) {
```

Comment:
we should move this functions to utils and import it, because I'm also using it in my component